tests/test_scan.py

```diff
@@ -365,3 +365,76 @@ def test_scan_execute_with_soda_server_client_scan_result_does_not_contain_any_e
     )
 
     assert not scan_result.has_errors()
+
+
+def test_scan_measurement_dataclass_fields(
+    scan_definition: str, df: DataFrame
+) -> None:
+    """Valid if the expected fields are present in measurement dataclass is present."""
+    key_list = []
+    scan_result = scan.execute(scan_definition, df)
+    for key in scan_result.measurements[0].__dataclass_fields__:
+        key_list.append(key)
+    assert key_list == ["metric", "column_name", "value", "group_values"]
```

Comment:
it would be fairer test if the schema in `scan.py` is used. You can either make the schema a global variable and import it here. Or compare the output of one of the `convert_` functions with data class

Reply:
I have reworked on the tests and defined new ones as discussed. Please review now.